### PR TITLE
 feat(Calender): [BLA-693] Adds Calendar component

### PR DESCRIPTION
**Pull request by swethab184**

### CODE REVIEW

File: packages/core-components/src/components/calendar/calender-days.tsx [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+  Prop,
+  State,
+  Listen,
+  Event,
+  EventEmitter,
+} from '@stencil/core';
+import {
+  DateSelectedEventDetail,
+  EscapePressed,
+} from '../../utils/interfaces/form.interface';
```
Comment (AnnaHoege) [inline packages/core-components/src/components/calendar/calender-days.tsx:15]: Should be in interaction interface as this is an interaction component.

File: packages/core-components/src/components/calendar/calender-days.scss [hunk 2]
```diff
@@ -0,0 +1,53 @@
+@use 'sass:color';
+@use '../../global/b2b-styles';
+
+.b2b-calender-days {
+  display: grid;
+  grid-template-columns: repeat(7, 1fr); /* Display days in a grid */
```
Comment (AnnaHoege) [inline packages/core-components/src/components/calendar/calender-days.scss:6]: should have a white background color

File: packages/core-components/src/components/calendar/calender-days.scss [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  padding-left: var(--b2b-size-40);
+
+  .b2b-calender-day {
```
Comment (AnnaHoege) [inline packages/core-components/src/components/calendar/calender-days.scss:9]: should also have a white background color

File: packages/core-components/src/components/calendar/calendar.scss [hunk 4]
```diff
@@ -0,0 +1,46 @@
+@use '../../global/b2b-styles';
+
+:host {
+  .b2b-calendar {
+    position: relative;
+  }
+
+  .b2b-calender-input-wrapper {
+    display: flex;
```
Comment (AnnaHoege) [inline packages/core-components/src/components/calendar/calendar.scss:9]: should have a white background color

File: packages/core-components/src/components/calendar/calender-days.scss [hunk 5]
```diff
@@ -0,0 +1,53 @@
+@use 'sass:color';
```
Comment (AnnaHoege) [inline packages/core-components/src/components/calendar/calender-days.scss:1]: typo in file name (calender -> calendar)

File: packages/core-components/src/components/calendar/calender-days.tsx [hunk 6]
```diff
@@ -0,0 +1,235 @@
+import {
```
Comment (AnnaHoege) [inline packages/core-components/src/components/calendar/calender-days.tsx:1]: typo in file name (calender -> calendar)

File: packages/core-components/src/components/calendar/calendar.scss [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  }
+
+  .b2b-calendar-label {
+    font-size: var(--b2b-size-35);
+    color: var(--b2b-color-black-100);
+    line-height: var(--b2b-size-copy-line-height-100);
+    margin-bottom: var(--b2b-size-5);
+  }
+
+  .b2b-datepicker {
```
Comment (AnnaHoege) [inline packages/core-components/src/components/calendar/calendar.scss:33]: why is this called datepicker?

File: packages/core-components/src/components/calendar/calendar.e2e.tsx [hunk 8]
```diff
@@ -0,0 +1,50 @@
+import { newE2EPage } from '@stencil/core/testing';
```
Comment (AnnaHoege) [inline packages/core-components/src/components/calendar/calendar.e2e.tsx:1]: would be good to also test disabled functionality and events

File: packages/core-components/src/components/calendar/calendar.tsx [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+                  aria-label={
+                    this.showCalendar ? 'close calender' : 'open calender'
+                  }
+                  icon="b2b_icon-event"
+                  clickable={true}></b2b-icon>
+              </div>
+            </div>
+          </div>
+        </div>
+        {this.showCalendar && (
```
Comment (AnnaHoege) [inline packages/core-components/src/components/calendar/calendar.tsx:172]: Personally, I would modify visibility via CSS here (display: block / display: none) as opposed to conditional rendering.

The current solution rerenders all dates again whenever the calendar is opened, which is less performant.

**Review by AnnaHoege — COMMENTED** [review]
Overall good work! I noticed some small things as listed below:

- add white background to all calendar files
- remove typos in file names
- add e2e tests for disabled test cases and test event emitting
- add comments to events (even if internal, you can use the @internal decorator)
- add visibility modifier to getPreviousMonth in calendar.tsx
- update screenshots with white background

File: packages/core-components/src/components/calendar/calendar.tsx [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+                    this.showHideCalendar();
+                  }}
+                  class="b2b-close-icon"
+                  onKeyDown={event => {
+                    if (event.key === 'Enter') {
+                      this.clearDateInput();
+                      this.showHideCalendar();
+                    }
+                  }}>
+                  <b2b-icon
```
[the remark this answers is not in the document]
Comment (swethab184) [reply packages/core-components/src/components/calendar/calendar.tsx:154]: will change after icon-100 release

**Pull request merged**